rcongui/src/components/PlayerInfo/PlayerInfo.js

```diff
@@ -0,0 +1,267 @@
+                open={open}
+                anchorEl={anchorEl}
+                onClose={handleClose}
+                anchorOrigin={{
+                    vertical: 'bottom',
+                    horizontal: 'center',
+                }}
+                transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'center',
+                }}
+            >
+                {names.map((name) => {
+                    return <Typography className="name">{name.name}</Typography>;
+                })}
+            </Popover>
+        </div>
+    );
+}
+
+const Punishment = ({punishments}) => {
+    const [myRowPerPage, setRowPerPage] = React.useState(
+        window.localStorage.getItem("punishments_row_per_page") || 50
+    );
+    const saveRowsPerPage = (rowPerPage) => {
+        window.localStorage.setItem("punishments_row_per_page", rowPerPage);
+        setRowPerPage(rowPerPage);
+    };
+    const columns = [
+        {name: "action_type", label: "Type"},
+        {name: "reason", label: "Reason"},
+        {name: "by", label: "By"},
+        {
+            name: "time",
+            label: "Time",
+            options: {
+                customBodyRenderLite: (dataIndex) =>
+                    moment(punishments[dataIndex].time).format("ddd Do MMM HH:mm:ss"),
+            },
+        }
+    ];
+
+    const options = {
+        filter: false,
+        rowsPerPage: myRowPerPage,
+        selectableRows: "none",
+        rowsPerPageOptions: [10, 25, 50, 100, 250, 500, 1000],
+        onChangeRowsPerPage: saveRowsPerPage,
+    };
+
+    return (
+        <MUIDataTable
+            title={"Punishments"}
+            data={punishments}
+            columns={columns}
+            options={options}
+        />
+    );
+}
+
+class PlayerInfo extends React.Component {
+    static propTypes = {
+        match: PropTypes.object.isRequired,
+        location: PropTypes.object.isRequired,
+        history: PropTypes.object.isRequired
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            steam_id_64: "",
+            created: "0",
+            names: [],
+            sessions: [],
+            sessions_count: 1086,
+            total_playtime_seconds: 4495950,
+            current_playtime_seconds: 17455,
+            received_actions: [],
+            penalty_count: {
+                TEMPBAN: 0,
+                PERMABAN: 0,
+                PUNISH: 0,
+                KICK: 0
+            },
+            blacklist: {
+                is_blacklisted: false,
+                reason: "",
+                by: ""
+            },
+            flags: [],
+            watchlist: {},
+            steaminfo: {},
+            perma: false,
+            temp: false,
+            vip: false,
+            loaded: false
+        }
+    }
+
+    /**
+     * fetch bans that currently affect the steamId64
+     * @param steamId64
+     */
+    #fetchPlayerBan = (steamId64) => {
```

Comment:
I'm not familiar with that # notation, what's that for?

Reply:
It s for private method but i removed it